src/api/bucket.ts

```diff
@@ -0,0 +1,144 @@
+import { Response } from 'node-fetch'
+import APIClient, { calculateAPIReset, getAPIOffset } from "./client";
+import Queue from "./queue";
+import APIRequest from "./request";
+
+export default class Bucket {
+  client: APIClient
+  queue: Queue = new Queue()
+  name: string
+  reset: number = -1
+  remaining: number = Infinity
+  limit: number = Infinity
+  retryAfter: number = -1
+
+  constructor (client: APIClient, name: string) {
+    this.client = client
+    this.name = name
+  }
+
+  get globalLimited (): boolean {
+    return this.client.globalRemaining <= 0
+      && this.client.globalReset !== undefined
```

Comment:
This is redundant with the `Date.now() < undefined` check

Reply:
Typescript forces doesn't allow you to do `< undefined`, should I replace it with nullish coalescing instead (`globalReset ?? 0`)?